lib/repo/subversion_repository.rb

```diff
@@ -565,44 +561,56 @@ def expand_path(file_name, dir_string = "/")
     ##  this class).
     ####################################################################
 
-    # Semi-private class method: Reads in Repository.conf[:REPOSITORY_PERMISSION_FILE]
+    # Semi-private class method
     def self.__read_in_authz_file
       # Check if configuration is in order
-      if Repository.conf[:REPOSITORY_PERMISSION_FILE].nil?
-        raise ConfigurationError.new("Required config 'REPOSITORY_PERMISSION_FILE' not set")
+      if !MarkusConfigurator.markus_config_repository_admin?
+        raise NotAuthorityError.new('Unable to read authsz file:' \
+                                    'Not in authoritative mode!')
+      end
+      if MarkusConfigurator.markus_config_repository_permission_file.nil?
+        raise ConfigurationError.new("Required config " \
+                                     "'REPOSITORY_PERMISSION_FILE' not set")
       end
-      if !File.exist?(Repository.conf[:REPOSITORY_PERMISSION_FILE])
-        File.open(Repository.conf[:REPOSITORY_PERMISSION_FILE], "w").close() # create file if not existent
+      if !File.exist?(MarkusConfigurator.
+                        markus_config_repository_permission_file)
+        # create file if it doesn't exist
+        File.open(MarkusConfigurator.
+                    markus_config_repository_permission_file, "w").close
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.